From: Cyrill Gorcunov <gorcunov@openvz.org>
Date: Tue, 21 Apr 2009 12:00:07 +0000 (+0400)
Subject: net: bridge - process skbs has been already substituted due to via_phys_dev
X-Git-Url: http://git.openvz.org/?p=linux-2.6.27-openvz;a=commitdiff_plain;h=ac9403982e62e2eb33ec19264d2d389f09034320

net: bridge - process skbs has been already substituted due to via_phys_dev

When via_phys_dev is enabled we substitute skb->dev with
master_dev and pass it back to bridge code. Instead of
dropping such skb we should pass it up to network stack
to process.

Signed-off-by: Cyrill Gorcunov <gorcunov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---

diff --git a/net/bridge/br_input.c b/net/bridge/br_input.c
index 44fb444..1a650ac 100644
--- a/net/bridge/br_input.c
+++ b/net/bridge/br_input.c
@@ -160,7 +160,7 @@ struct sk_buff *br_handle_frame(struct net_bridge_port *p, struct sk_buff *skb)
 		/* fall through */
 	case BR_STATE_LEARNING:
 		if (skb->brmark == BR_ALREADY_SEEN)
-			return 0;
+			return skb;
 
 		out = p->br->via_phys_dev ? p->br->master_dev : p->br->dev;
 
